From: Michael Chan <michael.chan@broadcom.com>
Date: Fri, 21 Apr 2017 20:11:25 -0400
Subject: bnxt_en: Check the FW_LLDP_AGENT flag before allowing DCBX host
 agent.
Patch-mainline: v4.12-rc1
Git-commit: 7d63818a35851cf00867248d5ab50a8fe8df5943
References: bsc#1020412 FATE#321671

Check the additional flag in bnxt_hwrm_func_qcfg() before allowing
DCBX to be done in host mode.

Signed-off-by: Michael Chan <michael.chan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/broadcom/bnxt/bnxt.c b/drivers/net/ethernet/broadcom/bnxt/bnxt.c
index 143b71a56589..e2c61949cf54 100644
--- a/drivers/net/ethernet/broadcom/bnxt/bnxt.c
+++ b/drivers/net/ethernet/broadcom/bnxt/bnxt.c
@@ -4460,7 +4460,8 @@ static int bnxt_hwrm_func_qcfg(struct bnxt *bp)
 	}
 #endif
 	if (BNXT_PF(bp) && (le16_to_cpu(resp->flags) &
-			    FUNC_QCFG_RESP_FLAGS_FW_DCBX_AGENT_ENABLED))
+			    (FUNC_QCFG_RESP_FLAGS_FW_DCBX_AGENT_ENABLED |
+			     FUNC_QCFG_RESP_FLAGS_FW_LLDP_AGENT_ENABLED)))
 		bp->flags |= BNXT_FLAG_FW_LLDP_AGENT;
 
 	switch (resp->port_partition_type) {
-- 
2.10.2

